server/routes/ad.ts

```diff
@@ -202,6 +202,10 @@ const getDetector = async (
         ? //@ts-ignore
           { stateError: detectorState.error }
         : {}),
+      ...(detectorState !== undefined
+        ? //@ts-ignore
+          { initProgress: getDetectorInitProgress(detectorState) }
+        : {}),
```

Comment:
Should we combine all 3 of these conditionals (setting detector `state`, `error`, `initProgress`) into one for readability?

Reply:
good catch. will do that.